num/quat/exp.go

```diff
@@ -0,0 +1,64 @@
+// Copyright ©2018 The Gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// Copyright 2017 The Go Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package quat
+
+import "math"
+
+// Exp returns e**q, the base-e exponential of q.
+func Exp(q Quat) Quat {
+	w, uv := split(q)
+	if uv == zero {
+		return lift(math.Exp(w))
+	}
+	v := Abs(uv)
+	e := math.Exp(w)
+	s, c := math.Sincos(v)
+	return join(e*c, Scale(e*s/v, uv))
+}
+
+// Log returns the natural logarithm of q.
+func Log(q Quat) Quat {
+	w, uv := split(q)
+	if uv == zero {
+		return lift(math.Log(w))
+	}
+	v := Abs(uv)
+	return join(math.Log(Abs(q)), Scale(math.Atan2(v, w)/v, uv))
```

Comment:
Where does the `Atan2` come from? The reference from Wikipedia https://users.aalto.fi/~ssarkka/pub/quat.pdf has `Acos`.

Reply:
That's a nice reference, thanks. I had misread the arguments.